controllers/che/backup.go

```diff
@@ -0,0 +1,84 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package che
+
+import (
+	"context"
+	"strings"
+
+	semver "github.com/blang/semver/v4"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	chev1 "github.com/eclipse-che/che-operator/api/v1"
+	"github.com/eclipse-che/che-operator/pkg/deploy"
+)
+
+func isPreviousVersionDeployed(cheCR *chev1.CheCluster) bool {
+	deployedVersion := cheCR.Status.CheVersion
+	currentVersion := deploy.DefaultCheVersion()
+	if deployedVersion == "" || deployedVersion == "next" || deployedVersion == currentVersion {
+		return false
+	}
+
+	deployedSemver, err := semver.Make(deployedVersion)
+	if err != nil {
+		logrus.Error(err)
```

Comment:
I would like to extend error message, like `It is not possible to parse a current version %s. Cause: %v`.
Maybe warning? Since we continue reconcile loop.

Reply:
Agree, will fix